<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1144</width>
    <height>700</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1000</width>
    <height>700</height>
   </size>
  </property>
  <property name="baseSize">
   <size>
    <width>1000</width>
    <height>700</height>
   </size>
  </property>
  <property name="font">
   <font>
    <family>Segoe UI</family>
    <pointsize>10</pointsize>
   </font>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="autoFillBackground">
   <bool>false</bool>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="CanvasWidget" name="canvasWidget" native="true">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>10</y>
      <width>821</width>
      <height>631</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(255, 255, 255);</string>
    </property>
   </widget>
   <widget class="QLabel" name="colorPreviewSolid">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>200</y>
      <width>51</width>
      <height>31</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(128, 64, 64);
</string>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="colorPreviewTransparent">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>200</y>
      <width>51</width>
      <height>31</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgba(128, 64, 64,1);
</string>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="label_Layers">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>350</y>
      <width>49</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Layers</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_ColorPreviewBackground">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>200</y>
      <width>51</width>
      <height>31</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(255, 255, 255);</string>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="label_ColorPreview">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>180</y>
      <width>81</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Color Preview</string>
    </property>
   </widget>
   <widget class="QPushButton" name="addLayerButton">
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>370</y>
      <width>71</width>
      <height>31</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;when adding layers, undo/redo history will be cleared.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="text">
     <string>add  </string>
    </property>
    <property name="icon">
     <iconset theme="QIcon::ThemeIcon::ListAdd"/>
    </property>
   </widget>
   <widget class="QPushButton" name="deleteLayerButton">
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>400</y>
      <width>71</width>
      <height>31</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;when deleting layers, undo/redo history will be cleared.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="text">
     <string>delete</string>
    </property>
    <property name="icon">
     <iconset theme="QIcon::ThemeIcon::ListRemove"/>
    </property>
   </widget>
   <widget class="QFrame" name="frame">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>50</y>
      <width>271</width>
      <height>71</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Shape::Box</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Shadow::Sunken</enum>
    </property>
    <widget class="QLabel" name="label_BrushSize">
     <property name="geometry">
      <rect>
       <x>180</x>
       <y>10</y>
       <width>31</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Size:</string>
     </property>
    </widget>
    <widget class="QSlider" name="brushOpacitySlider">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>161</width>
       <height>16</height>
      </rect>
     </property>
     <property name="maximum">
      <number>255</number>
     </property>
     <property name="value">
      <number>255</number>
     </property>
     <property name="sliderPosition">
      <number>255</number>
     </property>
     <property name="orientation">
      <enum>Qt::Orientation::Horizontal</enum>
     </property>
     <property name="invertedAppearance">
      <bool>false</bool>
     </property>
    </widget>
    <widget class="QLabel" name="brushSizeLabel">
     <property name="geometry">
      <rect>
       <x>210</x>
       <y>10</y>
       <width>21</width>
       <height>16</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>Segoe UI</family>
       <pointsize>11</pointsize>
       <underline>false</underline>
       <kerning>true</kerning>
      </font>
     </property>
     <property name="text">
      <string>20</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_BrushOpacity">
     <property name="geometry">
      <rect>
       <x>180</x>
       <y>29</y>
       <width>51</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Opacity:</string>
     </property>
    </widget>
    <widget class="QSlider" name="brushSizeSlider">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>161</width>
       <height>16</height>
      </rect>
     </property>
     <property name="maximum">
      <number>100</number>
     </property>
     <property name="sliderPosition">
      <number>20</number>
     </property>
     <property name="orientation">
      <enum>Qt::Orientation::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLabel" name="brushOpacityLabel">
     <property name="geometry">
      <rect>
       <x>230</x>
       <y>35</y>
       <width>31</width>
       <height>21</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>Segoe UI</family>
       <pointsize>11</pointsize>
       <underline>false</underline>
       <kerning>true</kerning>
      </font>
     </property>
     <property name="text">
      <string>100</string>
     </property>
    </widget>
   </widget>
   <widget class="QPushButton" name="layerUpButton">
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>430</y>
      <width>71</width>
      <height>41</height>
     </rect>
    </property>
    <property name="whatsThis">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;move layer up&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="text">
     <string>move
up</string>
    </property>
    <property name="icon">
     <iconset theme="QIcon::ThemeIcon::GoUp"/>
    </property>
   </widget>
   <widget class="QPushButton" name="layerDownButton">
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>470</y>
      <width>71</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>move
down</string>
    </property>
    <property name="icon">
     <iconset theme="QIcon::ThemeIcon::GoDown"/>
    </property>
   </widget>
   <widget class="QPushButton" name="pencilToolButton">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>130</y>
      <width>41</width>
      <height>41</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(169, 169, 169); border: none;
color: rgb(0, 0, 0);</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset>
      <normalon>:/resources/pencil-svgrepo-com.svg</normalon>
     </iconset>
    </property>
   </widget>
   <widget class="QPushButton" name="eraserToolButton">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>130</y>
      <width>41</width>
      <height>41</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: white ; border: none;
color: rgb(0, 0, 0);</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset resource="resources.qrc">
      <normaloff>:/resources/eraser-svgrepo-com.svg</normaloff>:/resources/eraser-svgrepo-com.svg</iconset>
    </property>
   </widget>
   <widget class="QListWidget" name="layersListWidget">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>370</y>
      <width>171</width>
      <height>271</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>Layer0</string>
     </property>
     <property name="checkState">
      <enum>Checked</enum>
     </property>
    </item>
   </widget>
   <widget class="QPushButton" name="undoButton">
    <property name="geometry">
     <rect>
      <x>60</x>
      <y>20</y>
      <width>31</width>
      <height>24</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;ctrl-z or ctrl-alt-z&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="styleSheet">
     <string notr="true"/>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset theme="QIcon::ThemeIcon::EditUndo"/>
    </property>
   </widget>
   <widget class="QPushButton" name="redoButton">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>20</y>
      <width>31</width>
      <height>24</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;ctrl-y or ctrl-shift-z&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="styleSheet">
     <string notr="true"/>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset theme="QIcon::ThemeIcon::EditRedo"/>
    </property>
   </widget>
   <widget class="QPushButton" name="saveButton">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>20</y>
      <width>31</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset resource="resources.qrc">
      <normaloff>:/resources/save-technology-svgrepo-com.svg</normaloff>:/resources/save-technology-svgrepo-com.svg</iconset>
    </property>
   </widget>
   <widget class="ColorPaletteButton" name="colorPalette1">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>250</y>
      <width>50</width>
      <height>31</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">border: none; 
background-color: rgb(138, 87, 240);</string>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="ColorPaletteButton" name="colorPalette2">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>250</y>
      <width>50</width>
      <height>31</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">border: none; 
background-color:rgb(112, 161, 215)</string>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="ColorPaletteButton" name="colorPalette3">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>250</y>
      <width>50</width>
      <height>31</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">border: none; 
background-color:rgb(161,222,147)</string>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="ColorPaletteButton" name="colorPalette4">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>250</y>
      <width>50</width>
      <height>31</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">border: none; 
background-color:rgb(247,244,139)</string>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="ColorPaletteButton" name="colorPalette5">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>250</y>
      <width>50</width>
      <height>31</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">border: none; 
background-color:rgb(244, 124, 124)</string>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="ColorPaletteButton" name="colorPalette6">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>280</y>
      <width>50</width>
      <height>31</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">border: none; 
background-color: rgb(95, 52, 128)</string>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="ColorPaletteButton" name="colorPalette10">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>280</y>
      <width>50</width>
      <height>31</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">border: none; 
background-color:rgb(177, 44, 33)</string>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="ColorPaletteButton" name="colorPalette7">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>280</y>
      <width>50</width>
      <height>31</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">border: none; 
background-color:rgb(0, 135, 205)</string>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="ColorPaletteButton" name="colorPalette8">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>280</y>
      <width>50</width>
      <height>31</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">border: none; 
background-color:rgb(78, 163, 42)</string>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="ColorPaletteButton" name="colorPalette9">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>280</y>
      <width>50</width>
      <height>31</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">border: none; 
background-color:rgb(249, 223, 0)</string>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QFrame" name="frame_2">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>240</y>
      <width>271</width>
      <height>80</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Shape::Panel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Shadow::Sunken</enum>
    </property>
   </widget>
   <zorder>frame_2</zorder>
   <zorder>label_ColorPreviewBackground</zorder>
   <zorder>canvasWidget</zorder>
   <zorder>colorPreviewSolid</zorder>
   <zorder>colorPreviewTransparent</zorder>
   <zorder>label_Layers</zorder>
   <zorder>label_ColorPreview</zorder>
   <zorder>addLayerButton</zorder>
   <zorder>deleteLayerButton</zorder>
   <zorder>frame</zorder>
   <zorder>layerUpButton</zorder>
   <zorder>layerDownButton</zorder>
   <zorder>pencilToolButton</zorder>
   <zorder>eraserToolButton</zorder>
   <zorder>layersListWidget</zorder>
   <zorder>undoButton</zorder>
   <zorder>redoButton</zorder>
   <zorder>saveButton</zorder>
   <zorder>colorPalette1</zorder>
   <zorder>colorPalette2</zorder>
   <zorder>colorPalette3</zorder>
   <zorder>colorPalette4</zorder>
   <zorder>colorPalette5</zorder>
   <zorder>colorPalette6</zorder>
   <zorder>colorPalette10</zorder>
   <zorder>colorPalette7</zorder>
   <zorder>colorPalette8</zorder>
   <zorder>colorPalette9</zorder>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1144</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>CanvasWidget</class>
   <extends>QWidget</extends>
   <header>canvaswidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ColorPaletteButton</class>
   <extends>QPushButton</extends>
   <header>colorpalettebutton.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections/>
</ui>
